### PR TITLE
Improve iterator and named arguments parsing

### PR DESCRIPTION
**Pull request by lukaszcz**
Avoid excessive backtracking in iterator and named arguments parsing. This also improves error messages by committing to a parsing branch as early as possible.

### CODE REVIEW

File: tests/positive/NamedArguments.juvix [hunk 1]
```diff
@@ -89,5 +89,5 @@ module FakeRecord;
     mkPair (snd := unit; fst := Type);
 
   pp2 : Pair (B := Unit; A := Type) :=
-    mkPair (fst := Type) unit;
+    mkPair (fst := Type) (unit);
```
Comment (jonaprieto) [inline tests/positive/NamedArguments.juvix:92]: Why this change?

Comment (lukaszcz) [reply tests/positive/NamedArguments.juvix:92]: To catch more errors. When it begins with `(` it can still be a range, you see it's not only later.

**Review by jonaprieto — APPROVED** [review]
LGTM
